code/src/main/java/com/google/sps/storage/data-storage/NegativeUserLocation.java

```diff
@@ -0,0 +1,54 @@
+package com.google.sps.storage;
+
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+import com.googlecode.objectify.annotation.Index;
+import java.time.Instant;
+import java.util.ArrayList;
+
+/**
+* This class stores the information needed to identify a negative user
+* at a specific location (geographic coordinate)
+*/
+@Entity
+public class NegativeUserLocation {
+
+  private String  userId;
+  @Index private long longitudeE7;
+  @Index private long latitudeE7;
+  @Index private long intervalStartSeconds;
+  @Index private long intervalEndSeconds;
+  @Id private Long key;
```

Comment:
the key here isn't initialized? You could store a UUID as the key (google UUID to learn about it, but it's basically a guarenteed unique string).

Reply:
I think I left a comment on DD for this one, I'll wait to check that out

Reply:
But to address your comment, the reason why I used Long instead of long is because Objectify will set it to a non null value when stores. So a UUID would have the same purpose if you prefer that.

Reply:
addressed